packages/opentelemetry/src/metrics.ts

```diff
@@ -91,3 +91,77 @@ export async function instrumentedWithMetrics<T>(
     histogram.record(performance.now() - start);
   }
 }
+
+export interface CreateObservableRangeGaugeOptions extends MetricOptions {
+  interval: number;
+}
+
+/**
+ * Creates a set of gauges that track the min, max, and average of a value over
+ * time. The value is observed on a regular interval.
+ *
+ * The provided {@link name} is used as the base name for the three gauges. The
+ * names of the individual gauges are:
+ *
+ * - `${name}.min`
+ * - `${name}.max`
+ * - `${name}.avg`
+ */
+export async function createObservableRangeGauge(
+  meter: Meter,
+  name: string,
+  options: CreateObservableRangeGaugeOptions,
+  observe: () => number
+) {
+  const { interval, ...metricOptions } = options;
+
+  let min = 0;
+  let max = 0;
+  let sum = 0;
+  let count = 0;
+
+  // Observe the value on a regular interval. Black-hole any errors.
+  const intervalId = setInterval(() => {
+    Promise.resolve(observe())
+      .then((value) => {
+        min = Math.min(min, value);
```

Comment:
With the initialized value `min = 0` we'll never update to any other value.
```suggestion
        min = (count == 0 ? value : Math.min(min, value));
```
Alternatively we could keep the existing update rule but initialize to `min = Number.MAX_VALUE` or `min = Infinity`. However, I worry how this will interact with gauges where the underlying type is `ValueType.INT` and so I think it's safer to keep the initialization as `min = 0` but change the update rule as above.

Reply:
Good catch! I like your solution here.